### ISSUE TITLE
ApiKey removed from v4.17.0

### ISSUE THREAD

**Issue by magreenbaum** [pulumi/pulumi-datadog#256]
### What happened?

The ApiKey module was removed from version 4.17.0. 

https://github.com/pulumi/pulumi-datadog/commit/4252bf9182dc8e0e68a922c608d8bfb0c33cdda1#diff-4d5a3babfd378d1a719d302bbe7bacf6af782219adb66edcd25f0a45bd8d5f30

```
Diagnostics:
  pulumi:pulumi:Stack (<stack_name>):
    error: Program failed with an unhandled exception:
    Traceback (most recent call last):
      File "<location>", line 22, in <module>
        datadog_api_key = datadog.ApiKey(
    AttributeError: module 'pulumi_datadog' has no attribute 'ApiKey'
```

### Expected Behavior

Apply ApiKey with no issues.

### Steps to reproduce

Upgrade to `pulumi_datadog` v4.17.0. 

### Output of `pulumi about`

```
CLI          
Version      3.65.1
Go Version   go1.20.3
Go Compiler  gc

Plugins
NAME     VERSION
datadog  4.14.0
```

### Contributing

Vote on this issue by adding a 👍 reaction. 
To contribute a fix for this issue, leave a comment (and link to your pull request, if you've opened one already).

**Comment by t0yv0** [comment]
Apologies, we missed the breaking change during routine upgrade. We have reverted the provider to the last known working version and this is now released as https://github.com/pulumi/pulumi-datadog/releases/tag/v4.17.1 - we'll be looking into the root cause so we can-redo the upgrade but without the breaking change. It looks like the root cause is that the affected resource and a few  data sources have moved to the Terraform Plugin Framework.
